Test/Remove-WhiskeyFileSystemItem.Tests.ps1

```diff
@@ -0,0 +1,208 @@
+& (Join-Path -Path $PSScriptRoot -ChildPath 'Initialize-WhiskeyTest.ps1' -Resolve)
+
+function GivenFileOrDirectory
+{
+    param(
+        $Path,
+
+        [string]$PathType
+    )
+
+    $fullPath = Join-Path -Path $TestDrive.FullName -ChildPath $Path
+    New-Item -Path $fullPath -ItemType $PathType -Force | Out-Null
+}
+

```

Comment:
Create an `Init` function that creates a per-test `$testRoot` variable. We can't use `TestDrive.FullName`. That directory is created for each `Describe` block. When we switch to the next version of Pester, we'll also be switching to one `Describe` block per file with multiple `It` blocks. These tests will fail when we do.

The solution is to create a test directory for each test. Do this in the `Init` function:

```powershell
function Init
{
    $script:testRoot = New-WhiskeyTestRoot
}
```

Update each `It` block to call `Init`. Replace all usages of `$TestDrive.FullName` with `$testRoot`. This will make migrating to Pester 5 easier.